lightning-c-bindings/src/ln/msgs.rs

```diff
@@ -0,0 +1,690 @@
+	/// " Handle an incoming commitment_signed message from the given peer."
+	pub handle_commitment_signed: extern "C" fn (this_arg: *const c_void, their_node_id: crate::c_types::PublicKey, msg: &CommitmentSigned),
+	/// " Handle an incoming revoke_and_ack message from the given peer."
+	pub handle_revoke_and_ack: extern "C" fn (this_arg: *const c_void, their_node_id: crate::c_types::PublicKey, msg: &RevokeAndACK),
+	/// " Handle an incoming update_fee message from the given peer."
+	pub handle_update_fee: extern "C" fn (this_arg: *const c_void, their_node_id: crate::c_types::PublicKey, msg: &UpdateFee),
+	/// " Handle an incoming announcement_signatures message from the given peer."
+	pub handle_announcement_signatures: extern "C" fn (this_arg: *const c_void, their_node_id: crate::c_types::PublicKey, msg: &AnnouncementSignatures),
+	/// " Indicates a connection to the peer failed/an existing connection was lost. If no connection"
+	/// " is believed to be possible in the future (eg they're sending us messages we don't"
+	/// " understand or indicate they require unknown feature bits), no_connection_possible is set"
+	/// " and any outstanding channels should be failed."
+	pub peer_disconnected: extern "C" fn (this_arg: *const c_void, their_node_id: crate::c_types::PublicKey, no_connection_possible: bool),
+	/// " Handle a peer reconnecting, possibly generating channel_reestablish message(s)."
+	pub peer_connected: extern "C" fn (this_arg: *const c_void, their_node_id: crate::c_types::PublicKey, msg: &Init),
+	/// " Handle an incoming channel_reestablish message from the given peer."
+	pub handle_channel_reestablish: extern "C" fn (this_arg: *const c_void, their_node_id: crate::c_types::PublicKey, msg: &ChannelReestablish),
+	/// " Handle an incoming error message from the given peer."
+	pub handle_error: extern "C" fn (this_arg: *const c_void, their_node_id: crate::c_types::PublicKey, msg: &ErrorMessage),
+	pub MessageSendEventsProvider: crate::util::events::MessageSendEventsProvider,
+}
+impl lightning::util::events::MessageSendEventsProvider for ChannelMessageHandler {
+	fn get_and_clear_pending_msg_events(&self) -> Vec<lightning::util::events::MessageSendEvent> {
+		unimplemented!()
+	}
+}
+unsafe impl Send for ChannelMessageHandler {}
+unsafe impl Sync for ChannelMessageHandler {}
+
+use lightning::ln::msgs::ChannelMessageHandler as lnChannelMessageHandler;
+impl lnChannelMessageHandler for ChannelMessageHandler {
+	fn handle_open_channel(&self, their_node_id: &bitcoin::secp256k1::key::PublicKey, their_features: lightning::ln::features::InitFeatures, msg: &lightning::ln::msgs::OpenChannel) {
+		(self.handle_open_channel)(self.this_arg, crate::c_types::PublicKey::from_rust(their_node_id), crate::ln::features::InitFeatures { inner: Box::into_raw(Box::new(their_features)) }, &OpenChannel { inner: msg })
```

Comment:
Fundamentally because there's a C equivalent (for a public key namely a 33 byte string). We could just serialize out InitFeatures into C (though its obviously more complicated since its not fixed-length), but then we'd lose things like "does_support_feature_X" (well, we dont have those now cause we cant run the macro to figure out the generated code, but at least for other types this holds).